lib/cldr/export/data/base.rb

```diff
@@ -53,11 +55,45 @@ def xpath(sources)
           end
 
           def doc
-            @doc ||= Nokogiri::XML(File.read(path))
+            @@doc_cache[paths.hash] ||= merge_paths(paths)
+          end
+
+          def paths
+            @paths ||= begin
+              if locale
+                Dir[File.join(Cldr::Export::Data.dir, "*", "#{Cldr::Export.from_i18n(locale)}.xml")].sort & Cldr::Export::Data.paths_by_root["ldml"]
+              else
+                Cldr::Export::Data.paths_by_root["supplementalData"]
+              end
+            end
           end
 
-          def path
-            @path ||= "#{Cldr::Export::Data.dir}/main/#{Cldr::Export.from_i18n(locale)}.xml"
+          private
+
+          def merge_paths(paths_to_merge)
+            # Some parts (`ldml`, `ldmlBCP47` amd `supplementalData`) of CLDR data require that you merge all the
+            # files with the same root element before doing lookups.
+            # Ref: https://www.unicode.org/reports/tr35/tr35.html#XML_Format
+            #
+            # The return of this method is a merged XML Nokogiri document.
+            # Note that it technically is no longer compliant with the CLDR `ldml.dtd`, since:
+            # * it has repeated elements
+            # * the <identity> elements no longer refer to the filename
+            #
+            # However, this is not an issue, since #select will find all of the matches from each of the repeated elements,
+            # and the <identity> elements are not important to us / make no sense when combined together.
+            return Nokogiri::XML('') if paths_to_merge.empty?
+
+            rest = paths_to_merge[1..paths_to_merge.size - 1]
```

Comment:
I can't use `paths_to_merge[1..]` since that's not available in Ruby 2.3, and we haven't officially dropped support for the old versions of Ruby.